documentation/versioned_docs/version-5.4/assertions/table_driven_testing.md

```diff
@@ -0,0 +1,90 @@
+# Table-driven testing
+
+> TK: what current exists is not really documented anywhere, am I right?
+
+## Define a table in code
+
+You can define a table of data that will be used for your test like this:
+
+```kotlin
+val table = table(
+  headers("id", "name", "username"),
+  row(4, "Jean-Michel Fayard", "jmfayard"),
+  row(6, "Louis CAD", "LouisCAD"),
+)
+```
+
+It's now easy to run your asserts for all rows of the table:
+
+## Run asserts forAll rows of the table
+
+```kotlin
+test("table-driven testing") {
+    table.forAll { id, name, username ->
+      id shouldBeGreaterThan 0
+      username shouldNotBe ""
+    }
+}
+```
+
+The test will not fail at the first error. Instead, it will always run on all the rows, and report multiple errors if they are present.
+
+Defining a table of data in code is convenient... until you start to have too much rows.
+
+## Export a table to a text file
+
+You can export your data to a text file with the `.table` extension.
+
+```kotlin
+val tableFile = testResources.resolve("users.table")
+table.writeTo(tableFile)
+```
+
+The `users.table` file looks like this:
+
+```csv
+id | username | fullName
+4  | jmfayard | Jean-Michel Fayard
+6  | louis    | Louis Caugnault
```

Comment:
What should happen if the user tries to load this table?
```csv
id | username | fullName
4  | jmfayard 
6  | louis    | Louis Caugnault
```

Reply:
```
AssertionError when parsing File("users.table")
headers("id | username | fullName") expect 3 columns
Got less than 3 columns at rows: [1, 42]
```

Reply:
I think I would allow and ignore empty lines or lines that start with `#` as a comment
https://stackoverflow.com/questions/1961006/can-a-csv-file-have-a-comment